icu4j/main/core/src/test/java/com/ibm/icu/dev/test/rbbi/RBBITestMonkey.java

```diff
@@ -1433,6 +1433,12 @@ void RunMonkey(BreakIterator  bi, RBBIMonkeyKind mk, String name, int  seed, int
                 if (c < 0) {   // TODO:  deal with sets containing strings.
                     errln("c < 0");
                 }
+                // Do not emit surrogates on Java 8, as the behaviour of regular expressions that
+                // match surrogates differs there.
+                if (System.getProperty("java.version").startsWith("1.") &&
+                        Character.isSurrogate((char)c)) {
```

Comment:
This also triggers for U+1D800, U+2D800, ...

Please check Character.isBmpCodePoint(c) first.

(Don't ask me why isSurrogate() does not take an int code point.)